homeassistant/helpers/device_registry.py

```diff
@@ -1169,6 +1255,7 @@ async def async_load(self) -> None:
         self.devices = devices
         self.deleted_devices = deleted_devices
         self._device_data = devices.data
+        self._merge_collisions()
```

Comment:
This should happen in a minor version bump instead if we can guarantee we won't introduce new collisions